Description: Move most helper binaries to /usr/lib/codelite
 Use /usr/bin/cppcheck as the path to cppcheck
Author: James Cowgill <james410@cowgill.org.uk>
Forwarded: not-needed
Last-Update: 2014-04-07
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/LiteEditor/clang_driver.cpp
+++ b/LiteEditor/clang_driver.cpp
@@ -777,7 +777,7 @@
 #ifdef __WXMAC__
     wxFileName exePath(wxStandardPaths::Get().GetDataDir(), wxT(""));
 #else
-    wxFileName exePath(wxStandardPaths::Get().GetExecutablePath());
+    wxFileName exePath(wxStandardPaths::Get().GetPluginsDir(), wxT(""));
 #endif
     exePath.SetFullName(wxT("codelite-clang"));
 
--- a/LiteEditor/manager.cpp
+++ b/LiteEditor/manager.cpp
@@ -1412,7 +1412,7 @@
     wd = bldConf->GetWorkingDirectory();
     wd = ExpandVariables ( wd, GetProject ( projectName ), clMainFrame::Get()->GetMainBook()->GetActiveEditor() );
 
-    wxFileName fnCodeliteTerminal(wxStandardPaths::Get().GetExecutablePath());
+    wxFileName fnCodeliteTerminal(wxStandardPaths::Get().GetPluginsDir(), wxT(""));
     fnCodeliteTerminal.SetFullName("codelite-terminal");
 
     wxString title;
@@ -1464,8 +1464,7 @@
             wxString command;
             if ( bldConf->GetPauseWhenExecEnds() ) {
                 wxString ld_lib_path;
-                wxFileName exePath ( wxStandardPaths::Get().GetExecutablePath() );
-                wxFileName exeWrapper ( exePath.GetPath(), wxT ( "codelite_exec" ) );
+                wxFileName exeWrapper ( wxStandardPaths::Get().GetPluginsDir(), wxT ( "codelite_exec" ) );
 
                 if ( wxGetEnv ( wxT ( "LD_LIBRARY_PATH" ), &ld_lib_path ) && ld_lib_path.IsEmpty() == false ) {
                     command << wxT ( "/bin/sh -f " ) << exeWrapper.GetFullPath() << wxT ( " LD_LIBRARY_PATH=" ) << ld_lib_path << wxT ( " " );
--- a/Plugin/globals.cpp
+++ b/Plugin/globals.cpp
@@ -1566,7 +1566,7 @@
     wxString title ( cmd );
 
     OptionsConfigPtr opts = EditorConfigST::Get()->GetOptions();
-    wxFileName fnCodeliteTerminal(wxStandardPaths::Get().GetExecutablePath());
+    wxFileName fnCodeliteTerminal(wxStandardPaths::Get().GetPluginsDir(), wxT(""));
     fnCodeliteTerminal.SetFullName("codelite-terminal");
 
     //change directory to the working directory
@@ -1599,8 +1599,7 @@
             // build the command
             wxString command;
             wxString ld_lib_path;
-            wxFileName exePath ( wxStandardPaths::Get().GetExecutablePath() );
-            wxFileName exeWrapper ( exePath.GetPath(), wxT ( "codelite_exec" ) );
+            wxFileName exeWrapper ( wxStandardPaths::Get().GetPluginsDir(), wxT ( "codelite_exec" ) );
 
             if ( wxGetEnv ( wxT ( "LD_LIBRARY_PATH" ), &ld_lib_path ) && ld_lib_path.IsEmpty() == false ) {
                 command << wxT ( "/bin/sh -f " ) << exeWrapper.GetFullPath() << wxT ( " LD_LIBRARY_PATH=" ) << ld_lib_path << wxT ( " " );
--- a/LiteEditor/frame.cpp
+++ b/LiteEditor/frame.cpp
@@ -931,9 +931,8 @@
     tagsManager->SetCodeLiteIndexerPath(wxStandardPaths::Get().GetDataDir());
 #else
     // set the path to codelite_indexer
-    wxFileName exePath( wxStandardPaths::Get().GetExecutablePath() );
-    tagsManager->SetCodeLiteIndexerPath(exePath.GetPath());
-    ManagerST::Get()->SetCodeLiteLauncherPath(exePath.GetPath());
+    tagsManager->SetCodeLiteIndexerPath(wxStandardPaths::Get().GetPluginsDir());
+    ManagerST::Get()->SetCodeLiteLauncherPath(wxStandardPaths::Get().GetPluginsDir());
 #endif
     tagsManager->StartCodeLiteIndexer();
 
--- a/Plugin/pipedprocess.cpp
+++ b/Plugin/pipedprocess.cpp
@@ -54,8 +54,7 @@
 {
 #ifdef __WXGTK__
 	wxString cmd;
-	wxFileName exePath(wxStandardPaths::Get().GetExecutablePath());
-	wxFileName script(exePath.GetPath(), wxT("codelite_kill_children"));
+	wxFileName script(wxStandardPaths::Get().GetPluginsDir(), wxT("codelite_kill_children"));
 	cmd << wxT("/bin/sh -f ") << script.GetFullPath() << wxT(" ") << GetPid();
 	wxExecute(cmd, wxEXEC_ASYNC);
 #else
--- a/Plugin/macros.h
+++ b/Plugin/macros.h
@@ -103,7 +103,7 @@
 
 // terminal macro
 #ifdef __WXGTK__
-#define TERMINAL_CMD  wxString::Format(wxT("%s/codelite_xterm '$(TITLE)' '$(CMD)'"), wxFileName(wxStandardPaths::Get().GetExecutablePath()).GetPath().c_str())
+#define TERMINAL_CMD  wxString::Format(wxT("%s/codelite_xterm '$(TITLE)' '$(CMD)'"), wxStandardPaths::Get().GetPluginsDir().c_str())
 #elif defined(__WXMAC__)
 #define TERMINAL_CMD  wxString::Format(wxT("%s/OpenTerm '$(CMD)'"), wxStandardPaths::Get().GetDataDir().c_str())
 #else
--- a/LiteEditor/app.cpp
+++ b/LiteEditor/app.cpp
@@ -576,8 +576,7 @@
 #endif
     wxString oldpath;
     wxGetEnv(wxT("PATH"), &oldpath);
-    wxFileName execfpath(wxStandardPaths::Get().GetExecutablePath());
-    wxSetEnv(wxT("PATH"), oldpath + pathsep + execfpath.GetPath());
+    wxSetEnv(wxT("PATH"), oldpath + pathsep + wxStandardPaths::Get().GetPluginsDir());
     wxString newpath;
     wxGetEnv(wxT("PATH"), &newpath);
 
--- a/Runtime/clg++
+++ b/Runtime/clg++
@@ -1,4 +1,3 @@
-#!/bin/bash
-codelitegcc g++ $@
-
+#!/bin/sh
+exec /usr/lib/codelite/codelitegcc g++ "$@"
 
--- a/Runtime/clgcc
+++ b/Runtime/clgcc
@@ -1,3 +1,3 @@
-#!/bin/bash
-codelitegcc gcc $@
+#!/bin/sh
+exec /usr/lib/codelite/codelitegcc gcc "$@"
 
--- a/CodeLite/procutils.cpp
+++ b/CodeLite/procutils.cpp
@@ -438,22 +438,11 @@
     if ( !programConsoleCommand.IsEmpty() ) {
         tokens = wxStringTokenize ( programConsoleCommand, wxT ( " " ), wxTOKEN_STRTOK );
         if ( !tokens.IsEmpty() ) {
-            configuredTerminal = wxStringTokenize ( tokens.Item(0), wxT ( "/" ), wxTOKEN_STRTOK );
-            if ( !configuredTerminal.IsEmpty() ) {
-                terminal = configuredTerminal.Last();
-                tokens.Clear();
-                configuredTerminal.Clear();
-            }
+			terminal = tokens.Item(0);
         }
     }
     if ( Locate ( terminal, where ) ) {
-        if ( terminal == wxT("konsole") ) {
-            wxString path = wxGetCwd();
-            terminal.Clear();
-            terminal << where << wxT(" --workdir \"") << path << wxT("\"");
-        } else {
-            terminal = where;
-        }
+		terminal = where;
     } else {
         return false;
     }
--- a/codelite_terminal/procutils.cpp
+++ b/codelite_terminal/procutils.cpp
@@ -438,22 +438,11 @@
     if ( !programConsoleCommand.IsEmpty() ) {
         tokens = wxStringTokenize ( programConsoleCommand, wxT ( " " ), wxTOKEN_STRTOK );
         if ( !tokens.IsEmpty() ) {
-            configuredTerminal = wxStringTokenize ( tokens.Item(0), wxT ( "/" ), wxTOKEN_STRTOK );
-            if ( !configuredTerminal.IsEmpty() ) {
-                terminal = configuredTerminal.Last();
-                tokens.Clear();
-                configuredTerminal.Clear();
-            }
+			terminal = tokens.Item(0);
         }
     }
     if ( Locate ( terminal, where ) ) {
-        if ( terminal == wxT("konsole") ) {
-            wxString path = wxGetCwd();
-            terminal.Clear();
-            terminal << where << wxT(" --workdir \"") << path << wxT("\"");
-        } else {
-            terminal = where;
-        }
+		terminal = where;
     } else {
         return false;
     }
--- a/cppchecker/cppchecker.cpp
+++ b/cppchecker/cppchecker.cpp
@@ -462,7 +462,7 @@
 
 #endif
 
-    path << wxFileName::GetPathSeparator() << wxT("codelite_cppcheck");
+    path << wxFileName::GetPathSeparator() << wxT("cppcheck");
 #ifdef __WXMSW__
     path << wxT(".exe");
 #endif
